tracks/typescript/exercises/resister-color-duo/mentoring.md

```diff
@@ -0,0 +1,47 @@
+
+### Reasonable Solutions
+
+```typescript
+export class ResistorColor {
+  
+  private readonly Colors: string[] = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
+  
+  private readonly MINIMUM_ACCEPTABLE_COLORS = 2;
+
+  constructor(private colors: string[]) {
+    
+    if (colors.length < this.MINIMUM_ACCEPTABLE_COLORS)
+      throw "At least two colors need to be present";
+
+  }
+
+  value = (): number => {
+    
+    const result = this.colors
+      .slice(0, this.MINIMUM_ACCEPTABLE_COLORS)
+      .reduce(
```

Comment:
This is just my opinion but I prefer a solution using `map`. i,e 
```
const result = this.colors
    .slice(0, this.MINIMUM_ACCEPTABLE_COLORS)
    .map((c) => this.Colors.indexOf(c))
    .join('')
```
I don't think using `reduce` is any less correct, it just it feels a bit of a strange use case for it here!

Happy to discuss!

Reply:
This works and is arguably much easier to read so I changed the example to do it this way. 👍